### PR TITLE
Go to beta for SOGo

### PR DESCRIPTION
**Pull request by stephdl**
Go to beta for ns8-sogo

For now the sogo is under my personal repository ghcr.io/stephdl/sogo, we could push it to forge, the repository is https://github.com/stephdl/docker-archlinux-sogo

link to https://github.com/NethServer/sogo-server/pull/1

image
image
image
image
image
image
image
image
image

https://github.com/NethServer/dev/issues/6804

### CODE REVIEW

File: build-images.sh [hunk 1]
```diff
@@ -36,10 +37,10 @@ buildah add "${container}" imageroot /imageroot
 buildah add "${container}" ui/dist /ui
 # Setup the entrypoint, ask to reserve one TCP port with the label and set a rootless container
 buildah config --entrypoint=/ \
-    --label="org.nethserver.authorizations=traefik@node:routeadm" \
+    --label="org.nethserver.authorizations=traefik@node:routeadm mail@any:mailadm" \
     --label="org.nethserver.tcp-ports-demand=1" \
     --label="org.nethserver.rootfull=0" \
-    --label="org.nethserver.images=docker.io/jmalloc/echo-server:latest" \
+    --label="org.nethserver.images=docker.io/mariadb:10.11.6 ghcr.io/stephdl/sogo:5.9.0" \
```
Comment (DavidePrincipi) [inline build-images.sh]: is it still the correct registry url?

File: imageroot/actions/create-module/20Env-default [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import agent
+import os
+from pathlib import Path
+
+# try to find the timezone except to UTC
+symlink_path = "/etc/localtime"
+resolved_path = Path(symlink_path).resolve()
+timezone = str(resolved_path.relative_to(Path("/usr/share/zoneinfo")))
+if not timezone:
+    timezone = "UTC"
```
Comment (DavidePrincipi) [inline imageroot/actions/create-module/20Env-default:19]: Does it work  with both Debian and EL flavors?

File: imageroot/templates/sogo.conf [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        UIDFieldName = sAMAccountName ;
+        IMAPLoginFieldName = sAMAccountName;
+        canAuthenticate = YES;
+        bindDN = "{{ldap_user}}";
+        bindPassword = "{{ldap_password}}";
+        baseDN = "{{ldap_base}}";
+        bindFields = (
+                sAMAccountName,
+        );
+        hostname = ldap://10.0.2.2:{{ldap_port}};
```
Comment (DavidePrincipi) [inline imageroot/templates/sogo.conf]: You could set `accountprovider` which is resolved by the container /etc/hosts file

File: imageroot/actions/create-module/20Env-default (hunk 2 above)
Comment (stephdl) [reply imageroot/actions/create-module/20Env-default:19]: yes tested on debian and centos like (rocky linux)

File: build-images.sh (hunk 1 above)
Comment (stephdl) [reply build-images.sh]: adjusted now

File: imageroot/templates/sogo.conf (hunk 3 above)
Comment (stephdl) [reply imageroot/templates/sogo.conf]: done

File: imageroot/update-module.d /20restart [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+set -e
+
+# Redirect any output to the journal (stderr)
+exec 1>&2
+
+# Restart if running, ignore if stopped
+systemctl --user try-restart sogo.service
```
Comment (DavidePrincipi) [inline imageroot/update-module.d /20restart:14]: Did you try the pod restart? Are *-app.service restarted too?

File: ui/public/metadata.json [hunk 5]
```diff
@@ -3,17 +3,17 @@
...
     {
-      "name": "Name Surname",
-      "email": "author@yourmail.org"
+      "name": "stephane de Labrusse",
+      "email": "stephdl@de-labrusse.fr"
     }
   ],
   "docs": {
-    "documentation_url": "https://docs.sogo.com/",
+    "documentation_url": "https://www.sogo.nu/files/docs/SOGoInstallationGuide.html",
```
Comment (DavidePrincipi) [inline ui/public/metadata.json]: Should be the link to the app manual page

File: imageroot/update-module.d /20restart (hunk 4 above)
Comment (stephdl) [reply imageroot/update-module.d /20restart:14]: normally we binds to sogo.service (pod), I verified but yes when we trigger a restart sogo.service then we restart sogo-app and mariadb-app 

https://github.com/NethServer/ns8-sogo/blob/goToBeta/imageroot/systemd/user/mariadb-app.service#L8

https://github.com/NethServer/ns8-sogo/blob/goToBeta/imageroot/systemd/user/sogo-app.service#L8

File: ui/public/metadata.json (hunk 5 above)
Comment (stephdl) [reply ui/public/metadata.json]: done

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**